store/repository.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	"github.com/odpf/siren/store/model"
+	"github.com/odpf/siren/store/postgres"
+	"gorm.io/gorm"
+)
+
+type NamespaceRepository interface {
+	Migrate() error
+	List() ([]*model.Namespace, error)
+	Create(*model.Namespace) (*model.Namespace, error)
+	Get(uint64) (*model.Namespace, error)
+	Update(*model.Namespace) (*model.Namespace, error)
+	Delete(uint64) error
```

Comment:
repository should accept and return `*domain.Namespace` instead of `*model.Namespace` so we don't have to use `namespaceModel.ToDomain()`, `namespaceModel.FromDomain()` or any other model operations inside service. It should be inside the repository only.
plus, when it's only used inside repository, `ToDomain()` and `FromDomain()` could be private methods as well